### PR TITLE
Refactor password field validation and restyle

### PR DESCRIPTION
**Pull request by lukaw3d**
I tried to make validation part of the input field, so that onSubmit doesn't need to re-check validity of every input field. But then error-specific-styling doesn't work `contentProps={{ border: props.error ? 'bottom' : false }}`.

| Before | After |
| --- | --- |
|  localhost_3000_open-wallet_private-key (2) | localhost_3000_open-wallet_private-key  |

### CODE REVIEW

File: src/app/components/PasswordField/index.tsx [hunk 1]
```diff
@@ -2,39 +2,45 @@ import { Box, FormField, Button, TextInput, Tip } from 'grommet'
...
+  label?: string
+  placeholder?: string
   inputElementId: string
 
   autoComplete: 'on' | 'off' | 'new-password' | 'current-password'
-  value: string
-  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
-  error: string | false
+  value?: string
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
```
Comment (buberdds) [inline src/app/components/PasswordField/index.tsx:13]: It looks like these type changes does not belong to this PR. I don't see any use cases for optional `value` and `onChange` or 'TFormValue' here.

Comment (lukaw3d) [reply src/app/components/PasswordField/index.tsx:13]: The new recommended usage is with `name` and `validate`, but `value`, `onChange`, `error` still come useful. And keeping them is consistent with grommet's FormField.

I added some tests to validate both uses

**Review by buberdds — APPROVED** [review]
